construct/core.py

```diff
@@ -1017,8 +1017,9 @@ class FormatField(Construct):
     def __init__(self, endianity, format):
         if endianity not in list("=<>"):
             raise FormatFieldError("endianity must be like: = < >", endianity)
-        if format not in list("fdBHLQbhlq"):
-            raise FormatFieldError("format must be like: f d B H L Q b h l q", format)
+
+        if format not in list("fdBHLQbhlqe"):
+            raise FormatFieldError("format must be like: f d B H L Q b h l q e", format)
         super(FormatField, self).__init__()
         self.fmtstr = endianity+format
         self.length = struct.calcsize(endianity+format)
```

Comment:
This line calculates the size for "e" regardless of Python version, thus exception. You could just put an IF on this line, that would solve the problem.